### DISCUSSION TITLE
Record Audio track only during video call

### DISCUSSION THREAD

**Discussion by sidpram** [ant-media/Ant-Media-Server#6448] [Q&A]
Enhance recording API to either record Audio and Video or only the Audio track in a video call.

When we start recording a call via API, is it possible to specify to record only Audio, even if the call is a video call?

your suggestion or solution is highly appreciated. 

Thanks,
Siddharth

**Comment by burak-58** [comment]
Hi @sidpram,
Let me convert this issue a github discussion so that you get faster suggestions.

**Comment by yashtandon113** [comment]
Hi @sidpram 

Once you start the stream recording with video and audio. After the stream is finished, you can initiate the user defined script that will remove the video from the recording using FFMPEG and save the audio only. Please follow below steps to test it.

1. Create one script with name for example, `removevideo.sh` in any of the folders, like `/home/ubuntu/`.

```
#create the bash script to remove video from the recorded file and save it with same id again

# Don't forget to change the Ant Media Server App Name
AMS_APP_NAME="LiveApp"

file="$1"
temp_file="${file%.mp4}_temp.mp4"

cd /usr/local/antmedia/$AMS_APP_NAME/LiveApp/streams/

# Add metadata using ffmpeg
ffmpeg -i "$file" -c copy -vn "$temp_file"

# Replace the original file with the new file
mv "$temp_file" "$file"
```

2. Provide the execute permission to the script using below command.

`sudo chmod +x removevideo.sh
`

3. After that, go to application's advanced settings and set the path of script to the below property.

`  "muxerFinishScript": "/home/ubuntu/removevideo.sh",
`

4. Publish the stream on the server and after the stream is stopped, the video will be removed from the recording. If the script executes successfully after stopping the stream, then you should see the log line below.

`2024-07-02 20:53:29,777 [vert.x-worker-thread-86] INFO  i.a.AntMediaApplicationAdapter - completing script: /home/ubuntu/removevideo.sh  /usr/local/antmedia/webapps/LiveApp/streams/test.mp4 with return value 0`


I hope this fulfills your requirement. Please feel free to ask if you have any questions.
